en/docs/references/troubleshoot-choreo.md

```diff
@@ -67,3 +67,43 @@ To resolve the issue, follow the guidelines given below:
       For step-by-step instructions on how to manage users with Choreo's built-in identity provider (IdP), see [Configure a User Store with the Built-In IdP](../administer/configure-a-user-store-with-built-in-idp.md).
       
       For details on setting up other OpenID Connect (OIDC) supported IdPs, see [Manage OAuth Keys](../authentication-and-authorization/secure-web-applications-with-managed-authentication/#step-3-manage-oauth-keys).
+
+## Troubleshoot Tailscale proxy issues
+
+- ### Where can I find logs to troubleshoot Tailscale proxy issues?
+
+      To troubleshoot Tailscale proxy issues, you can view the [Runtime Logs](https://wso2.com/choreo/docs/monitoring-and-insights/view-logs/#runtime-logs) of the running container for your Tailscale proxy deployment. These logs can help you diagnose most of the issues. 
+      You can also view real-time container logs via the **Runtime** page under **DevOps**. For more details, see [Observe real-time container logs](https://wso2.com/choreo/docs/devops-and-ci-cd/view-runtime-details/#observe-real-time-container-logs).
+
+- ### I'm not able to connect the Tailscale proxy node to my Tailscale network due to an authentication failure.
+
+      - If you encounter the following log lines in your Tailscale proxy deployment, it indicates a misconfiguration of the `TS_AUTH_KEY`:
+
+         ```
+         2024-06-04T10:38:53.885800940Z To authenticate, visit:
+         2024-06-04T10:38:53.885802684Z 
+         2024-06-04T10:38:53.885815708Z https://login.tailscale.com/a/696841f011517
+         2024-06-04T10:38:53.885817457Z 
+         2024-06-04T10:38:55.194344862Z Waiting for tailscale up to complete...
+         2024-06-04T10:38:57.198970796Z Waiting for tailscale up to complete...
+         2024-06-04T10:38:59.203265659Z Waiting for tailscale up to complete...
+         ```
+        To resolve this, you must re-check your authentication key and ensure you have entered the correct key.
+
+      - If you encounter the following log lines in your Tailscale proxy deployment, it indicates that your authentication key is invalid or expired.
+         ```
+         2024-06-04T11:33:58.762363181Z 2024/06/04 11:33:58 Received error: invalid key: unable to validate API key
+         2024-06-04T11:33:58.762458209Z backend error: invalid key: unable to validate API key
+         ```
+       
+         To resolve this, you must verify the correctness of your authentication key. If the key has expired, you must generate a new key from Tailscale admin console. 
+
+
+- ### I'm not able to access private endpoints although the Tailscale proxy is properly connected to my Tailscale network.
+
+      To resolve this, do the following:
+
+       - Ensure your on-premises setup is properly connected to the Tailscale network and that specific services and database servers are running as expected in your on-premises setup.
```

Comment:
```suggestion
       - Ensure your on-premises setup is properly connected to the Tailscale network and that specific services, database servers, etc.. are running as expected in your on-premises setup.
```

Reply:
Applied the suggestion with a minor change.
We generally try to avoid using `etc.` in the middle of a sentence, but if it has to be used in the middle of a sentence, it is followed by a comma [1].
[1] https://developers.google.com/style/word-list#etc